acceptance/tests/vault/vault_namespaces_test.go

```diff
@@ -0,0 +1,182 @@
+package vault
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	terratestLogger "github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/environment"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/k8s"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/vault"
+	"github.com/stretchr/testify/require"
+)
+
+// TestVault installs Vault, configures a Vault namespace, and then bootstraps it
+// with secrets, policies, and Kube Auth Method.
+// It then configures Consul to use vault as the backend and checks that it works
+//with the vault namespace.
+func TestVault_VaultNamespace(t *testing.T) {
+	cfg := suite.Config()
+	ctx := suite.Environment().DefaultContext(t)
+	ns := ctx.KubectlOptions(t).Namespace
+	vaultNamespacePath := "test-namespace"
+	consulReleaseName := helpers.RandomName()
+	vaultReleaseName := helpers.RandomName()
+
+	k8sClient := environment.KubernetesClientFromOptions(t, ctx.KubectlOptions(t))
+	vaultLicenseSecretName := fmt.Sprintf("%s-enterprise-license", vaultReleaseName)
+	vaultLicenseSecretKey := "license"
+
+	vaultEnterpriseLicense := os.Getenv("VAULT_ENT_LICENSE")
+
+	logger.Log(t, "Creating secret for Vault license")
+	consul.CreateK8sSecret(t, k8sClient, cfg, ns, vaultLicenseSecretName, vaultLicenseSecretKey, vaultEnterpriseLicense)
+	vaultHelmvalues := map[string]string{
+		"server.image.repository":             "hashicorp/vault-enterprise",
```

Comment:
should we use a mirror here to prevent dockerhub rate limiting? We hopefully won't hit it because it's just this one test, but maybe as a precaution and for consistency.
```suggestion
		"server.image.repository":             "docker.mirror.hashicorp.services/hashicorp/vault-enterprise",
```